src/components/utils/src/file_system_qt.cc

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (c) 2014, Ford Motor Company
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "utils/file_system.h"
+#include "utils/logger.h"
+#include "utils/string_utils.h"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sstream>
+#include <QtCore>
+#include <QStorageInfo>
+#include <QDir>
+#include <QDebug>
+#include <QFileInfo>
+#include <QUrl>
+
+#include <io.h>
+// TODO(VS): lint error: Streams are highly discouraged.
+#include <fstream>
+#include <cstddef>
+#include <algorithm>
+
+#define R_OK    4
+#define W_OK    2
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
+
+uint64_t file_system::GetAvailableDiskSpace(const std::string& path) {
+ QStorageInfo * mstor = new QStorageInfo(QString(path.c_str()));
+ qint64 b_aval = mstor->bytesAvailable();
+ if (b_aval != 0) {
+    return static_cast<uint64_t>(b_aval);
+  } else {
+    return 0;
+  }
+}
+
+int64_t file_system::FileSize(const std::string &path) {
+  if (file_system::FileExists(path)) {
+    return static_cast<uint64_t>(QFileInfo(QString(path.c_str())).size());
+  }
+  return 0;
+}
+
+size_t file_system::DirectorySize(const std::string& path) {
+    quint64 size = 0;
+     QFileInfo str_info(QString(path.c_str()));
+     if (str_info.isDir()){
+         QDir dir(QString(path.c_str()));
+         QFileInfoList list = dir.entryInfoList(QDir::Files | QDir::Dirs |  QDir::Hidden | QDir::NoSymLinks | QDir::NoDotAndDotDot);
+         for (int i = 0; i < list.size(); ++i) {
```

Comment:
Personally I don't like Qt in general. If we would use c++11 ever, then we will have native foreach... Lets wote ;)